tests/base/symmetric_tensor_39.cc

```diff
@@ -0,0 +1,70 @@
+#include <iostream>
+#include <fstream>
+#include <type_traits>
+
+#include <deal.II/base/symmetric_tensor.h>
+#include <deal.II/base/vectorization.h>
+
+using namespace dealii;
+
+int main()
+{
+  std::string logname = "output";
+  std::ofstream logfile(logname);
+
+#if defined DEAL_II_COMPILER_VECTORIZATION_LEVEL
+  std::cout << "Vectorization preprocessor variable found!" << std::endl;
+#endif
+
+#if DEAL_II_COMPILER_VECTORIZATION_LEVEL == 0
+  std::cout << "Value of DEAL_II_COMPILER_VECTORIZATION_LEVEL represents no vectorization" << std::endl;
+#elif DEAL_II_COMPILER_VECTORIZATION_LEVEL == 1
+  std::cout << "Value of DEAL_II_COMPILER_VECTORIZATION_LEVEL represents SSE" << std::endl;
+#elif DEAL_II_COMPILER_VECTORIZATION_LEVEL == 2
+  std::cout << "Value of DEAL_II_COMPILER_VECTORIZATION_LEVEL represents AVX" << std::endl;
+#else
+  std::cout << "Value of DEAL_II_COMPILER_VECTORIZATION_LEVEL not in {0,1,2}" << std::endl;
+#endif
+
+  const unsigned int dim = 3;
+
+  SymmetricTensor<4,dim,VectorizedArray<double>> I;
+  SymmetricTensor<2,dim,VectorizedArray<double>> A;
+  SymmetricTensor<2,dim,VectorizedArray<double>> B;
+
+  // I^sym = 0.5(d_ik*d_jl + d_il*d_jk) -> I^sym : A = A^sym
+  for (unsigned int i = 0; i < dim; i++)
+    for (unsigned int j = 0; j < dim; j++)
+      for (unsigned int k = 0; k < dim; k++)
+        for (unsigned int l = 0; l < dim; l++)
+          I[i][j][k][l] = ( (i == k && j== l && i == l && j == k) ? make_vectorized_array(1.0) : ( (i == k && j== l) || (i == l && j == k) ? make_vectorized_array(0.5) : make_vectorized_array(0.0)));
+
+  double counter = 0.0;
+  for (unsigned int i = 0; i < dim; ++i)
+    for (unsigned int j = 0; j < dim; ++j)
+      for (unsigned int v = 0; v < VectorizedArray<double>::n_array_elements; v++)
+        {
+          A[i][j][v] = counter;
+          counter += 1.0;
+        }
+
+  B = I*A;
+  B -= A;
+
+  // Note that you cannot use B.norm() here even with something
+  // like VectorizedArray<double> frob_norm = B.norm() -> Maybe a TODO?
+  for (unsigned int i = 0; i < dim; ++i)
+    for (unsigned int j = 0; j < dim; ++j)
+      for (unsigned int v = 0; v < VectorizedArray<double>::n_array_elements; ++v)
+        if (B[i][j][v] != 0.0)
+          logfile << "Not Ok" << std::endl;
+
+  if (std::is_pod<VectorizedArray<double> >::value)
```

Comment:
please move this and next into a stand-alone test, you also need to mark the `output` file as `with_cxx11=on` as those functions require c++11.